src/core/frames/frame_controller.ts

```diff
@@ -163,7 +164,16 @@ export class FrameController implements AppearanceObserverDelegate, FetchRequest
 
   formSubmissionSucceededWithResponse(formSubmission: FormSubmission, response: FetchResponse) {
     const frame = this.findFrameElement(formSubmission.formElement)
-    frame.delegate.loadResponse(response)
+    const target = [ formSubmission.formElement, formSubmission.submitter, frame ]
+      .map(element => element?.getAttribute("data-turbo-frame"))
+      .find(target => target)
+
+    if (response.redirected && target == "_top") {
+      navigator.formSubmission = formSubmission
```

Comment:
@sstephenson this felt like the shortest distance to achieve the outcome. The awkwardness signals to me that there is a missing concept or delegate here.

A means of delegating or directly accessing the Navigator would also enable the transformation of a frame navigation to a fully formed `Visit` to drive the page.